p2p/conn/connection.go

```diff
@@ -550,7 +559,24 @@ FOR_LOOP:
 		var err error
 		_n, err = cdc.UnmarshalBinaryLengthPrefixedReader(c.bufConnReader, &packet, int64(c._maxPacketMsgSize))
 		c.recvMonitor.Update(int(_n))
+
 		if err != nil {
+			// connection is closed (likely by the other side)
+			// shut down our side
+			if err == io.EOF {
+				c.stopServices()
```

Comment:
You shouldn't have to call this separately. `c.Stop()` calls `c.OnStop()` which calls `stopServices` for us.

Reply:
Removed.